tasktiger/redis_scripts.py

```diff
@@ -506,6 +522,22 @@ def get_expired_tasks(self, key_prefix, time, batch_size, client=None):
         # [queue1, task1, queue2, task2] -> [(queue1, task1), (queue2, task2)]
         return list(zip(result[::2], result[1::2]))
 
+    def truncated_rpush(self, key, element, max_length, client=None):
```

Comment:
Just use a pipeline with `rpush` and `ltrim -max_length -1`.

Reply:
bfd5b7b6da67beec141dd277f539f96f50b2ce09